docs/contributor/image_mode.md

```diff
@@ -0,0 +1,217 @@
+ARG USER_PASSWD=\$5\$XDVQ6DxT8S5YWLV7\$8f2om5JfjK56v9ofUkUAwZXTxJl3Sqnc9yPnza4xoJ0
+RUN useradd -m -d /var/home/redhat -G wheel -p ${USER_PASSWD} redhat
+
+# Configure OpenShift pull secret
+ARG PULL_SECRET
+RUN echo $PULL_SECRET > /etc/crio/openshift-pull-secret && \
+    chmod 600 /etc/crio/openshift-pull-secret
+
+# Configure firewall
+RUN firewall-offline-cmd --zone=public --add-port=22/tcp && \
+    firewall-offline-cmd --zone=trusted --add-source=10.42.0.0/16 && \
+    firewall-offline-cmd --zone=trusted --add-source=169.254.169.1 && \
+    firewall-offline-cmd --zone=public --add-port=80/tcp && \
+    firewall-offline-cmd --zone=public --add-port=443/tcp && \
+    firewall-offline-cmd --zone=public --add-port=30000-32767/tcp && \
+    firewall-offline-cmd --zone=public --add-port=30000-32767/udp
+
+# Recursively make the root filesystem subtree shared as required by OVN images
+CMD ["/bin/sh", "-c", "mount --make-rshared / ; exec /sbin/init"]
+```
+
+> **Important:**<br>
+> When building a container image, podman uses host subscription information and
+> repositories inside the container. With only `BaseOS` and `AppStream` system
+> repositories enabled by default, we need to enable the `rhocp` and `fast-datapath`
+> repositories for installing MicroShift and its dependencies. These repositories
+> must be accessible in the host subscription, but not necessarily enabled on the host.
+
+Run the following image build command to create a local `bootc` image.
+
+Note that a pull secret file is used for:
+* The podman `--authfile` argument to pull the base `rhel-bootc:9.4` image from the `registry.redhat.io` registry
+* The build `PULL_SECRET` argument to be saved as `/etc/crio/openshift-pull-secret` file on the image for pulling MicroShift container images
+
+```
+PULL_SECRET=~/.pull-secret.json
+IMAGE_NAME=microshift-4.16-bootc
+
+sudo podman build --authfile "${PULL_SECRET}" -t "${IMAGE_NAME}" \
+    --build-arg PULL_SECRET="$(cat "${PULL_SECRET}")" \
+    -f Containerfile
+```
+
+Verify that the local MicroShift 4.16 `bootc` image was created.
+
+```
+$ sudo podman images "${IMAGE_NAME}"
+REPOSITORY                       TAG         IMAGE ID      CREATED        SIZE
+localhost/microshift-4.16-bootc  latest      193425283c00  2 minutes ago  2.31 GB
+```
+
+### Run Container
+
+Run the following command to start the MicroShift `bootc` image in an interactive
+terminal session. Note that the host container storage path is shared with the
+image so that it is reused in subsequent runs.
+
+```
+sudo podman run --rm -it --privileged \
+    -v /var/lib/containers/storage:/var/lib/containers/storage \
+    --name "${IMAGE_NAME}" \
+    "${IMAGE_NAME}"
+```
+
+Log into the container host when prompted using the `redhat:redhat` credentials.
+Run the following commands to start the MicroShift service and verify that all
+the pods are up and running without errors.
+
+```
+sudo systemctl start microshift
```

Comment:
We don't need this step because in our container file we are already enabling `microshift` service so it is going to start eventually. 
```
● microshift.service - MicroShift
     Loaded: loaded (/usr/lib/systemd/system/microshift.service; enabled; preset: disabled)
     Active: activating (start) since Wed 2024-07-31 11:49:12 UTC; 23s ago
   Main PID: 585 (microshift)
      Tasks: 19 (limit: 1638)
     Memory: 215.6M
        CPU: 10.931s
     CGroup: /system.slice/microshift.service
             └─585 microshift run
```